src/test/scala/za/co/absa/commons/reflect/ReflectionUtilsSpec.scala

```diff
@@ -37,37 +39,72 @@ class ReflectionUtilsSpec extends AnyFlatSpec with Matchers {
     plus(Map("x" -> 2, "y" -> 40)) should be(42)
   }
 
-  "extractProperties()" should "for given Product return a map of element names to their values" in {
+  behavior of "extractProperties()"
+
+  it should "for given Product return a map of element names to their values" in {
     // case class
     ReflectionUtils.extractProperties(Foo("aaa")) should be(Map("x" -> "aaa", "y" -> 42))
     ReflectionUtils.extractProperties(Foo("aaa", 777)) should be(Map("x" -> "aaa", "y" -> 777))
     // normal class
     ReflectionUtils.extractProperties(new Bar("bbb", 42)) should be(Map("a" -> "bbb"))
   }
 
-  "extractFieldValue()" should "return a value of a private field" in {
+  behavior of "extractFieldValue()"
+
+  it should "return a value of a private field of a Scala object" in {
     ReflectionUtils.extractFieldValue[Int](Foo, "privateVal") should be(42)
   }
 
-  "extractFieldValue()" should "return values of compiler generated private fields" in {
+  it should "return a value of a private field of a Java object" in {
+    ReflectionUtils.extractFieldValue[Array[Char]]("foo", "value") should equal("foo".toCharArray)
+  }
+
+  it should "return values of compiler generated private fields" in {
     val bar = new Bar("Pi", 3.14)
     ReflectionUtils.extractFieldValue[String](bar, "a") shouldEqual "Pi"
     ReflectionUtils.extractFieldValue[Double](bar, "b") shouldEqual 3.14
   }
 
-  "extractFieldValue()" should "extract from a field declared in any of the superclasses" in {
+  it should "extract from a field declared in any of the superclasses" in {
     class SubBar(a: String, b: Double) extends Bar(a, b)
     val subSubBar = new SubBar("Pi", 3.14) {}
     ReflectionUtils.extractFieldValue[String](subSubBar, "a") shouldEqual "Pi"
     ReflectionUtils.extractFieldValue[Double](subSubBar, "b") shouldEqual 3.14
   }
 
-  "extractFieldValue()" should "extract from a field using a provided class tag" in {
+  it should "extract from a field using a provided class tag" in {
     val subBar = new Bar("Pi", 3.14) {}
     ReflectionUtils.extractFieldValue[Bar, String](subBar, "a") shouldEqual "Pi"
   }
 
-  "ModuleClassSymbolExtractor" should "extract objects" in {
+  it should "extract from a private field used in a lambda" in {
+    class A(x: Int) { () => x }
```

Comment:
Just out of a curiosity. How does this lambda affect fetching the field?

Reply:
It's due to the lambda encoding. For that code the Scala compiler generates a "hidden" field to store the constructor parameter `x` value. The field has a so called _expanded_ name and looks like `za.co.absa.commons.reflect.ReflectionUtilsSpec$$anonfun$8$A$2.x`

I won't pretend I fully comprehend the reasoning behind that, especially because adding `def y = x` to that class above would result in the field name being just `x` instead of that beast, but that's what we've got.